src/processing.ts

```diff
@@ -177,3 +177,93 @@ export const min = (a: bigint, b: bigint) => (a < b ? a : b);
  * const biggest = max(5n, 10n) // 10n
  */
 export const max = (a: bigint, b: bigint) => (a > b ? a : b);
+
+/**
+ * Register the proposals as as active (to enable later identification
+ * of answers) and issue notifications for them.
+ *
+ * @param space - The space that the proposals belong to
+ * @param proposals - The proposals to process
+ *
+ * @example
+ * await processProposals(space, proposals)
+ */
+export const processProposals = (space: Space, proposals: ProposalQuestionCreated[]) => {
+  return configurableProcessProposals({
+    space,
+    proposals,
+    notifyFn: notify,
+    insertProposalFn: insertProposal,
+  });
+};
+
+type ConfigurableProcessProposalsDeps = {
+  space: Space;
+  proposals: ProposalQuestionCreated[];
+  notifyFn: typeof notify;
+  insertProposalFn: typeof insertProposal;
+};
+export const configurableProcessProposals = async (deps: ConfigurableProcessProposalsDeps) => {
+  const { space, proposals, insertProposalFn, notifyFn } = deps;
+
+  await Promise.all(
+    proposals.map((event) =>
+      Promise.all([
+        insertProposalFn({
+          ens: space.ens,
+          proposalId: event.proposalId,
+          txHash: event.txHash,
+          questionId: event.questionId,
+          happenedAt: event.happenedAt,
+        }),
+        notifyFn({
+          type: EventType.PROPOSAL_QUESTION_CREATED,
+          event,
+          space,
+        }),
+      ]),
+    ),
+  );
+};
+
+/**
+ * Notifies the answers, ignoring those not issued for the current space
+ *
+ * @param space - The space that the answers belong to
+ * @param answers - The answers to process
+ *
+ * @example
+ * await processAnswers(space, answers)
+ */
+export const processAnswers = (space: Space, answers: LogNewAnswer[]) => {
+  return configurableProcessAnswers({
+    space,
+    answers,
+    notifyFn: notify,
+    findProposalByQuestionIdFn: findProposalByQuestionId,
+  });
+};
+
+type ConfigurableProcessAnswersDeps = {
+  space: Space;
+  answers: LogNewAnswer[];
+  notifyFn: typeof notify;
+  findProposalByQuestionIdFn: typeof findProposalByQuestionId;
+};
+export const configurableProcessAnswers = async (deps: ConfigurableProcessAnswersDeps) => {
+  const { space, answers, notifyFn } = deps;
+
+  await Promise.all(
+    answers.map(async (event) => {
+      const proposal = await deps.findProposalByQuestionIdFn(event.questionId);
```

Comment:
Optional
can you destructure findProposalByQuestionIdFn from deps on line 254

Reply:
@divyangchauhan Completely agree: that is the pattern present in all other places in code, this doesn't have a reason to be different.

Reply:
@divyangchauhan done